database/sqlite.go

```diff
@@ -0,0 +1,73 @@
+package database
+
+import (
+	"errors"
+
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+type SqliteDatabase struct {
+	*gorm.DB
+}
+
+func NewSqliteDatabase(path string) *SqliteDatabase {
+	db, err := gorm.Open(sqlite.Open(path), &gorm.Config{
+		Logger: logger.Default.LogMode(logger.Silent),
+	})
+	if err != nil {
+		// TODO handle database error
+		panic(err)
+	}
+	db.AutoMigrate(&Entry{})
+	return &SqliteDatabase{db}
+}
+
+func (db *SqliteDatabase) CreateEntry(entry *Entry) error {
+	if entry.Name == "" {
+		return ErrorEntryEmptyName
+	}
+	if entry.Path == "" {
+		return ErrorEntryEmptyPath
+	}
+	return db.Create(entry).Error
+}
+
+func (db *SqliteDatabase) UpdateEntry(entry *Entry, column string, value interface{}) error {
+	if column == "id" {
+		return ErrorEntryCantUpdateID
+	}
+	if column == "name" && value == "" {
```

Comment:
Is there a pattern or package that makes it easy to set validation rules? I could see this getting more complex

Reply:
im pretty sure there's a way with gorm to set validation rules, but i didn't look into it, maybe we can implement it later and stick to this for now